### ISSUE TITLE
GPU Memory Optimization Needed

### ISSUE THREAD

**Issue by dodufish** [kwuking/TimeMixer#143]
memory usage is too high!! when training larger models it breaks 💥

## problem
- big batch sizes lead to OOM errors on standard GPUs
- no gradient accumulation support
- no memory-efficient attention used 

## fix suggestion
add gradient accumulation steps parameter to reduce memory:
```python
# pseudo-code
args.gradient_accumulation = 4  # accumulate over 4 steps

for i, (batch_x, ...) in enumerate(data_loader):
    # Forward pass
    outputs = model(batch_x, ...)
    loss = criterion(outputs, targets) / args.gradient_accumulation
    
    # Backward pass
    loss.backward()
    
    if (i + 1) % args.gradient_accumulation == 0:
        optimizer.step()
        optimizer.zero_grad()
```

would enable training bigger models and longer sequences without needing bigger GPU 🚀~~~

**Comment by kevinliu2000** [comment]
Thank you for your advice. This support has been implemented with the Torch framework. If we have the following requirement, we can just use the framework. We don't have a takeaway from this.
